packages/@aws-cdk-testing/framework-integ/test/aws-kinesisfirehose/test/integ.iceberg-destination.ts

```diff
@@ -0,0 +1,151 @@
+          type: 'string',
+        },
+        {
+          name: 'data',
+          type: 'string',
+        },
+      ],
+    },
+  },
+});
+table.addDependency(database);
+
+// Create delivery stream with Iceberg destination using catalogArn
+const deliveryStreamWithCatalog = new firehose.DeliveryStream(stack, 'IcebergDeliveryStreamWithCatalog', {
+  destination: new firehose.IcebergDestination(bucket, {
+    catalogConfiguration: {
+      catalogArn: cdk.Stack.of(stack).formatArn({
+        service: 'glue',
+        resource: 'catalog',
+      }),
+    },
+    destinationTableConfigurations: [
+      {
+        databaseName: database.ref,
+        tableName: table.ref,
+        uniqueKeys: ['id'],
+      },
+    ],
+  }),
+});
+deliveryStreamWithCatalog.node.addDependency(table);
+
+// Create delivery stream with warehouse location
+const deliveryStreamWithWarehouse = new firehose.DeliveryStream(stack, 'IcebergDeliveryStreamWithWarehouse', {
+  destination: new firehose.IcebergDestination(bucket, {
+    catalogConfiguration: {
+      warehouseLocation: `s3://${bucket.bucketName}/warehouse`,
+    },
+    destinationTableConfigurations: [
+      {
+        databaseName: database.ref,
+        tableName: table.ref,
+        uniqueKeys: ['id'],
+      },
+    ],
+  }),
+});
+deliveryStreamWithWarehouse.node.addDependency(table);
+
+// Create delivery stream with custom role
+const customRole = new iam.Role(stack, 'CustomRole', {
+  assumedBy: new iam.ServicePrincipal('firehose.amazonaws.com'),
+});
+
+const deliveryStreamWithCustomRole = new firehose.DeliveryStream(stack, 'IcebergDeliveryStreamWithCustomRole', {
+  destination: new firehose.IcebergDestination(bucket, {
+    catalogConfiguration: {
+      catalogArn: cdk.Stack.of(stack).formatArn({
+        service: 'glue',
+        resource: 'catalog',
+      }),
+    },
+    destinationTableConfigurations: [
+      {
+        databaseName: database.ref,
+        tableName: table.ref,
+        uniqueKeys: ['id'],
+      },
+    ],
+    role: customRole,
+  }),
+});
+deliveryStreamWithCustomRole.node.addDependency(table);
+
+// Create delivery stream with all properties
+const deliveryStreamAllProperties = new firehose.DeliveryStream(stack, 'IcebergDeliveryStreamAllProperties', {
+  destination: new firehose.IcebergDestination(bucket, {
+    catalogConfiguration: {
+      catalogArn: cdk.Stack.of(stack).formatArn({
+        service: 'glue',
+        resource: 'catalog',
+      }),
+      warehouseLocation: `s3://${bucket.bucketName}/warehouse`,
+    },
+    destinationTableConfigurations: [
+      {
+        databaseName: database.ref,
+        tableName: table.ref,
+        uniqueKeys: ['id'],
+        s3ErrorOutputPrefix: 'errors/iceberg_table/',
+      },
+    ],
+    appendOnly: true,
+    bufferingInterval: cdk.Duration.minutes(5),
+    bufferingSize: cdk.Size.mebibytes(8),
+    retryDuration: cdk.Duration.hours(2),
+  }),
+});
+deliveryStreamAllProperties.node.addDependency(table);
```

Comment:
You shouldn't need this line because `table.ref` sets an [implicit dependency](https://docs.aws.amazon.com/AWSCloudFormation/latest/TemplateReference/aws-attribute-dependson.html) automatically (applicable for all of the above).